crates/publish/src/bindle_pusher.rs

```diff
@@ -4,20 +4,17 @@ use anyhow::{Context, Result};
 use bindle::{standalone::StandaloneRead, Id};
 use std::path::Path;
 
-type BindleClient = bindle::client::Client<spin_loader::bindle::BindleTokenManager>;
-
 /// Pushes a standalone bindle to a Bindle server.
 pub async fn push_all(
     path: impl AsRef<Path>,
     bindle_id: &Id,
-    client: &BindleClient,
-    server_url: &str,
+    bindle_connection_info: crate::BindleConnectionInfo,
 ) -> Result<()> {
     let reader = StandaloneRead::new(&path, bindle_id).await?;
     reader
-        .push(client)
+        .push(&bindle_connection_info.client()?)
```

Comment:
@itowlson Are you envisioning something like this?
```
        .push(
            &bindle_connection_info.client()
            .context(
                format!(
                    "Failed to create a bindle client from provided connection info '{}'",
                    bindle_connection_info.display()
                )
            )?
        )
        .await
        .with_context(|| push_failed_msg(path, &bindle_connection_info.base_url))
```

(Where, if so, or if on the right track, I'd add a `display()` function to BindleConnectionInfo for printing)

Reply:
Yeah, that sort of thing.  Though for readability I would probably pull it out into e.g.

```
let client = &bci.client().with_context(...);
...
.push(&client)
```

rather than having the massive `with_context` block breaking up the flow

Reply:
@itowlson updated in https://github.com/fermyon/spin/pull/407/commits/55c3f9515c357fff7df761a87942d9cb95112bb8 -- how does this look?  Happy to tweak some more...

Reply:
(I thought it best not to print the entire bci struct as it may contain sensitive info like the un/pw values for basic auth)

Reply:
Looks great!  Thanks!